yourgov/components/FormMobileFoodVendorPermit/task/FormStep7.tsx

```diff
@@ -1,72 +1,210 @@
 
-	const typeCorrectedErrors = errors as ShallowErrors<Step7FormSchema>;
+	const [modalIsVisible, setModalIsVisible] = useState(false);
+	const [showRemovedEmployeeMessage, setShowRemovedEmployeeMessage] =
+		useState(false);
+	const [showErrorMessage, setShowErrorMessage] = useState(false);
+
+	const addedEmployee = query.success
+		? step7State?.employee &&
+		  step7State.employee.find((employee) => employee?.id === query.success)
+		: undefined;
+	const [showAddedEmployeeMessage, setShowAddedEmployeeMessage] = useState(
+		!!addedEmployee
+	);
+	useEffect(() => {
+		setShowAddedEmployeeMessage(!!addedEmployee);
+	}, [addedEmployee]);
+
+	const closeModal = () => {
+		setShowAddedEmployeeMessage(false);
+		setEmployeeToRemove(undefined);
+	};
+
+	useEffect(() => {
+		setModalIsVisible(!!employeeToRemove);
+		setShowRemovedEmployeeMessage(false);
+	}, [employeeToRemove]);
+
+	const removeEmployee = () => {
+		step7SetState({
+			...step7GetState,
+			completed: false,
+			employee:
+				step7State?.employee &&
+				step7State.employee.filter(
+					(employee) => employee?.id !== employeeToRemove?.id
+				),
+		});
+
+		setRemovedEmployee(
+			`${employeeToRemove?.firstName} ${employeeToRemove?.lastName}`
+		);
+		closeModal();
+
+		setTimeout(() => {
+			setShowRemovedEmployeeMessage(true);
+		}, 0);
+	};
+
+	const onSubmit: FormEventHandler<HTMLFormElement> = async (event) => {
+		event.preventDefault();
 
-	const onSubmit: SubmitHandler<Step7FormSchema> = async (data) => {
 		if (isSavingBeforeExiting) {
 			return;
 		}
+
+		if (step7State?.employee && step7State?.employee.length === 0) {
+			setShowRemovedEmployeeMessage(false);
+			setShowErrorMessage(true);
+			return;
+		}
+
 		await submitStep();
 		step7SetState({
-			...data,
+			...step7State,
 			completed: !isSavingBeforeExiting,
 			started: true,
 		});
 	};
 
 	return (
 		<FormContainer
+			formIntroduction="Add your employee details."
 			formTitle="Employees"
-			formIntroduction="Add details of any employees who will be handling food."
+			shouldFocusTitle={!showAddedEmployeeMessage}
 		>
-			<Stack as="form" gap={3} onSubmit={handleSubmit(onSubmit)} noValidate>
-				<Controller
-					control={control}
-					name="cuisine"
-					render={({ field: { ref, ...field } }) => (
-						<ComboboxMulti
-							label="TODO"
-							hint="TODO"
-							inputRef={ref}
-							options={cuisineOptions}
-							required
-							{...field}
-							id="cuisine"
-							invalid={Boolean(typeCorrectedErrors.cuisine?.message)}
-							message={typeCorrectedErrors.cuisine?.message}
-							maxWidth="xl"
-							block={false}
-						/>
-					)}
-				/>
-				<StepActions />
+			<Stack gap={2}>
+				{showErrorMessage && (
+					<PageAlert focusOnMount title="No employees added" tone="error">
+						<Text as="p">
+							You need to add some staff. You’re amazing, but you can’t do this
+							alone. Life is better with friends and team mates.
```

Comment:
😆 

This is fun but is it what we're really going with?

Reply:
I was as surprised as you…